b2share/modules/records/cli.py

```diff
@@ -102,6 +104,49 @@ def check_and_update_handle_records(update, verbose):
                         click.secho('  file PID ok: {}'.format(pid))
 
 
+@b2records.command()
+@with_appcontext
+@click.option('-u', '--update', is_flag=True, default=False)
+@click.argument('record_id', required=True)
+def check_handles(update, record_id):
+    """Allocates handles for a record and its files, if necessary"""
+    rec_pid = RecordUUIDProvider.get(pid_value=record_id).pid
+    record = Record.get_record(rec_pid.object_uuid)
+    record_updated = False
+
+    pid_list = [p.get('value') for p in record['_pid']
+                if p.get('type') == 'ePIC_PID']
+    if pid_list:
+        click.secho('record {} already has a handle'.format(record_id), fg='green')
+    else:
+        click.secho('record {} has no handle'.format(record_id), fg='red')
+        if update:
+            b2share_pid_minter(rec_pid, record)
+            record_updated = True
+            click.secho('    handle added to record', fg='green')
+        else:
+            click.secho('use -u argument to add a handle to the record')
+
+    files_ok = True
+    for f in record.get('_files', []):
```

Comment:
We should check what happens here for external files referenced by PID.

Reply:
https://github.com/EUDAT-B2SHARE/b2share/issues/1658

Reply:
https://github.com/EUDAT-B2SHARE/b2share/pull/1659